ietf-microwave-radio-link.yang

```diff
@@ -121,6 +126,45 @@ module ietf-microwave-radio-link {
        for configuration or status.";
   }
 
+  grouping header-compression-profile {
+  leaf header-compression-profile-name {
+    type string;
+    description
+      "none";
+  }
+  description
+    "Header compression profile: can be a list of protocol header to be compressed
+     or a header size in which all included (also partially) headers get compressed.";
+    choice header-compression {
+      case protocol-based {
+        leaf-list compressed-protocol-layer-list {
+          type mw-types:protocol-layer;
+          default "protocol-layer-type-none";
+          config false;
+          description
+            "List of protocol layers that get compressed with this kind of header compression.";
+        }
+        leaf-list mpls-payload-kind-list {
+          type mw-types:mpls-payload;
+          default "mpls-palyload-type-none";
+          config false;
+          description
+            "Defines how the header bytes inside the MPLS header has to be interpreted.";
+        }
+      }
+      case header-length-based {
+          leaf compressed-header-length {
```

Comment:
Adding an header-length-based container would be more future proof (allowing also proprietary augmentations) but not strictly needed to check which modes are supporting header-length-based header compression and which modes are not, since there is only one leaf which can be used instead of the container